code-challenges.test.js

```diff
@@ -14,45 +14,134 @@
+    })
+  })
 
-const filterLetterA = "a"
-// Expected output: ["Mango", "Apricot", "Peach"]
-const filterLetterE = "e"
-// Expected output: ["Cherry", "Blueberry", "Peach"]
 
 // b) Create the function that makes the test pass.
 
 // Pseudo code:
 
+// INPUT: an array of words, and a string with a letter
+// OUTPUT: and array of words in the string that correspond with a passed in letter
+// FUNCTION: name filterIt with 2 parameters array and letter
+// PROCESS:
+// 1) create a function that takes will be able to take in 2 arguments
+// 2) map through the first parameter
+// 3) create a conditional statement if value inclueds the 2nd parameter return it
+// 4) since this will return the entire array with true & unidentifed values we filter the outcome to return only boolean values. BadaBimBadaBoom!!!!
+
+const filterIt = (array, letter) => {
+    return array.map((value) => {
+        if (value.toLowerCase().includes(letter) === true ){
+            return value
+        } 
+    }).filter(Boolean)
+}
+
 // --------------------3) Create a function that takes in an array of 5 numbers and determines whether or not the array is a "full house". A full house is exactly one pair and one three of a kind.
 
 // a) Create a test with expect statements using the variable provided.
 
-const hand1 = [5, 5, 5, 3, 3]
-// Expected output: true
-const hand2 = [5, 5, 3, 3, 4]
-// Expected output: false
-const hand3 = [5, 5, 5, 5, 4]
-// Expected output: false
-const hand4 = [7, 2, 7, 2, 7]
-// Expected output: true
+
+describe("letsMakeItUnreadable", () => {
+    
+    const hand1 = [5, 5, 5, 3, 3]
+    // Expected output: true
+    const hand2 = [5, 5, 3, 3, 4]
+    // Expected output: false
+    const hand3 = [5, 5, 5, 5, 4]
+    // Expected output: false
+    const hand4 = [7, 2, 7, 2, 7]
+    // Expected output: true
+    
+  
+    it("in an array of 5 numbers and determines whether or not the array is a 'full house'", () => {
+      expect(fullHouse(hand1)).toEqual(true)
+      expect(fullHouse(hand2)).toEqual(false)
+      expect(fullHouse(hand3)).toEqual(false)
+      expect(fullHouse(hand4)).toEqual(true)
+    })
+  })
 
 // b) Create the function that makes the test pass.
 
 // Pseudo code:
+
+// INPUT: array of numbers
+// OUTPUT: if the argument is true =true, and if not = false
+// FUNCTION: name fullHouse with parameter array
+// PROCESS:
+// 1) create a function that takes in one parameter, an array
+// 2) create a code block that will take in the count
+// 3) create a for loop that will itirate through the array
+// 4) create a conditional statement if value = value add to counter the number of times is found it
+// 5) else the one time just return that one time occuered occurance
+// 6) create new Variable = birings object with counted times of each value
+// 7) return array when object includes 2 && 3
+
+const fullHouse = (arr) => {
+    const counts = {};
```

Comment:
Your pseudo code says you are creating a code block here, but you are creating an empty object. The function is then iterating through the array, and creating a key of the current value of the iteration in the object. If the key does not exist yet, we are creating a key and setting the value equal to 1. If the key already exists, then we are just adding 1 to the current value. This effectively stores a count of how many times each key appeared in the array.